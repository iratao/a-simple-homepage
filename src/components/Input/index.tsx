import React from 'react'
import styles from './index.css'

interface Props {
  value?: string;
  placeholder: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input = ({ placeholder, onChange, value }: Props) => {
  return (
    <input className={styles.input} placeholder={placeholder} onChange={onChange} value={value}/>
  )
}

export default Input;