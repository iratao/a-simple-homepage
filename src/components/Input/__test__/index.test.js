import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../index'

it('renders correctly', () => {
  const tree = renderer
    .create(<Input />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('correctly shows custom placeholder', () => {
  const tree = renderer
    .create(<Input placeholder='custom input placeholder'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})

it('can be set with value', () => {
  const tree = renderer
    .create(<Input value='custom input value'/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
})