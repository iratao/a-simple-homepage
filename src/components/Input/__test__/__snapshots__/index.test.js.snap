// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be set with value 1`] = `
<input
  className="input"
  value="custom input value"
/>
`;

exports[`correctly shows custom placeholder 1`] = `
<input
  className="input"
  placeholder="custom input placeholder"
/>
`;

exports[`renders correctly 1`] = `
<input
  className="input"
/>
`;
